File: 010.c
// 142913828922
#include <stdio.h>
#include <math.h>

int is_prime(int n)
{
    for (int i = 2; i < sqrt(n) + 1; i++) {
        if (n % i == 0) {
            return 0;
        }
    }

    return 1;
}

int main()
{
    long long int sum = 2;
    int limit = 2000000;

    for (int i = 2; ; i++) {
        if (is_prime(i)) {
            if (i > limit) {
                break;
            }

            sum += i;
        }
    }

    printf("%ld\n", sum);
    return 0;
}

File: zig/010.zig
const std = @import("std");
const stdout = std.io.getStdOut().writer();

fn isPrime(n: u64) bool {
    var nf: f64 = @floatFromInt(n);
    var limit: u64 = @intFromFloat(nf);
    for (2..limit) |i| {
        if (@mod(n, i) == 0) {
            return false;
        }
    }

    return true;
}

fn solve(n: u64) u64 {
    var sum: u64 = 0;

    for (2..n) |i| {
        if (isPrime(i)) {
            sum += i;
        }
    }

    return sum;
}

pub fn main() !void {
    const n = 2000000;
    try stdout.print("{}\n", .{solve(n)});
}
